corehq/apps/custom_data_fields/edit_model.py

```diff
@@ -59,7 +59,7 @@ class XmlSlugField(forms.SlugField):
     default_validators = [
         validate_slug,
         validate_reserved_words,
-        RegexValidator(r'^[a-zA-Z]', ''),
+        RegexValidator(r'\D', ''),  # disallow property names that are just numbers, which breaks
```

Comment:
I'm confused by both the before and after here - does `RegexValidator` raise an error if the value doesn't match the string?  Ie, would the old version succeed as long as there is any alphabetic character?  Does the new one succeed as long as there is any non numeric character present?  What about leading digits and invalid characters?

We do have this function which checks for valid xml identifiers:
https://github.com/dimagi/commcare-hq/blob/619c956550cd1409678163ff22083338e891ccd2/corehq/apps/fixtures/utils.py#L21-L21
A pretty robust approach would be to wrap that:
```python
def validate_xml_identifier(slug):
    if is_identifier_invalid(slug):
        raise ValidationError("message")
```

Reply:
Oh, `validate_slug` might take care of invalid characters.  Awesome new line linking there, github!

Reply:
Curious what'll happen if I link a code block:
https://github.com/dimagi/commcare-hq/blob/619c956550cd1409678163ff22083338e891ccd2/corehq/apps/fixtures/utils.py#L21-L37